bitnami/airflow/templates/worker-statefulsets.yaml

```diff
@@ -0,0 +1,309 @@
+            - name: git-cloned-dag-files
+              mountPath: /dags
+            {{- if $.Values.airflow.cloneDagFilesFromGit.sshKeySecretName }}
+            - name: dag-git-keys
+              mountPath: /root/.ssh
+            {{- end }}
+        {{- end }}
+        {{- if .additionalRepos.enabled }}
+        {{- range .additionalRepos.repos }}
+        - name: git-clone-{{ .name | camelcase | kebabcase}}
+          image: "{{ template "git.image" $ }}"
+          imagePullPolicy: {{ $.Values.git.pullPolicy | quote }}
+          command:
+            - /bin/bash
+            - -ec
+            - |
+              git clone {{ .repository }} --branch {{ .branch }} /{{ .name }}
+          volumeMounts:
+            - name: git-cloned-{{ .name }}
+              mountPath: /{{ .name }}
+            {{- if .sshKeySecretName }}
+            - name: {{ .name }}-git-keys
+              mountPath: /root/.ssh
+            {{- end }}
+        {{- end }}
+        {{- end }}
+      containers:
+        {{- if $.Values.airflow.cloneDagFilesFromGit.enabled }}
+        - name: dag-repo-syncer
+          image: "{{ template "git.image" $ }}"
+          imagePullPolicy: {{ $.Values.git.pullPolicy | quote }}
+          command:
+            - /bin/bash
+            - -ec
+            - |
+              while true; do
+                  cd /dags && git pull origin {{ $.Values.airflow.cloneDagFilesFromGit.branch }}
+                  sleep {{ default "60" $.Values.airflow.cloneDagFilesFromGit.interval }}
+              done
+          volumeMounts:
+            - name: git-cloned-dag-files
+              mountPath: /dags
+            {{- if $.Values.airflow.cloneDagFilesFromGit.sshKeySecretName }}
+            - name: dag-git-keys
+              mountPath: /root/.ssh
+            {{- end }}
+        {{- end }}
+        {{- if .additionalRepos.enabled }}
+        {{- range .additionalRepos.repos }}
+        {{- if .syncFilesFromGit }}
+        - name: {{ .name }}-repo-syncer
+          image: "{{ template "git.image" $ }}"
+          imagePullPolicy: {{ $.Values.git.pullPolicy | quote }}
+          command:
+            - /bin/bash
+            - -ec
+            - |
+              while true; do
+                  cd /{{ .name }} && git pull origin {{ .branch }}
+                  sleep {{ default "60" .interval }}
+              done
+          volumeMounts:
+            - name: git-cloned-{{ .name }}
+              mountPath: /{{ .name }}
+            - name: {{ .name }}-git-keys
+              mountPath: /root/.ssh
+        {{- end }}
+        {{- end }}
+        {{- end }}
+        {{- else }}
+      containers:
+      {{- end }}
+      - name: {{ .name | camelcase | kebabcase }}
+        {{- if .image }}
+        image: "{{ .image.registry }}/{{ .image.repository }}:{{ .image.tag }}"
+        imagePullPolicy: {{ .image.pullPolicy default "IfNotPresent" }} 
+        {{- else }}
+        image: "docker.io/bitnami/airflow-worker:1.10.6-debian-9-r1"
```

Comment:
We have some chart update automations in place that may not work correctly with this approach. How important is it that each worker has a different image? Maybe we can have that the different workers have the same image and only the configuration is the different thing. That would make the chart more maintainable and potentially more usable. Does it make sense?

Reply:
The entire idea of my pull request was to allow the user to provide fully customized workers. So you could specify your own image, own configurations, everything. This is because the default bitnami images are very basic and provide no additional libraries. For instance, I want to be able to use [papermill](https://papermill.readthedocs.io/en/latest/) with airflow, so I need to add it manually to the image. The ability to have multiple workers, each with different images is to allow for each worker to be dedicated to its individual task.